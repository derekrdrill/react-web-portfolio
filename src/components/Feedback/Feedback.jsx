import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { FeedbackList } from './components/FeedbackList';
import { FeedbackStats } from './components/FeebackStats';
import { FeedbackForm } from './components/FeedbackForm';
import { FeedbackProvider } from './context/FeedbackContext';

export const Feedback = () => (
  <>
    <PageBodyStyle />
    <FeedbackProvider>
      <AppContainer>
        <FeedbackForm />
        <FeedbackStats />
        <FeedbackList />
      </AppContainer>
    </FeedbackProvider>
  </>
);

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: '#202142',
    padding: 20,
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
  },
});

const AppContainer = styled.div({
  maxWidth: 1100,
  margin: 'auto',
  padding: '0 20px',
});
