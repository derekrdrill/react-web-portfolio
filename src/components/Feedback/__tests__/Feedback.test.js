import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Feedback from '../Feedback';

const renderer = new ShallowRenderer();

describe('Feedback tests', () => {
  it('renders correctly', () => {
    renderer.render(<Feedback />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
