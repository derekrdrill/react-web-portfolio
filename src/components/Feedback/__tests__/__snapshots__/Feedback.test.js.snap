// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback tests renders correctly 1`] = `
<React.Fragment>
  <Memo(l) />
  <FeedbackProvider>
    <styled.div>
      <FeedbackForm
        maxRank={10}
      />
      <FeedbackStats />
      <FeedbackList />
    </styled.div>
  </FeedbackProvider>
</React.Fragment>
`;
